cache.go

```diff
@@ -545,7 +545,7 @@ func (c *Cache) doWithRetries(ctx context.Context, r *request) (*http.Response,
 }
 
 func (c *Cache) url(p string) string {
-	return c.URL + "_apis/artifactcache/" + p
+	return strings.TrimRight(c.URL, "/") + "/_apis/artifactcache/" + p
```

Comment:
Would it make sense to sanitize / normalize the URL field as part of the constructor, or are there other ways the field can be / is expected to be set?

Reply:
I was thinking of parsing the URL with `url.Parse` and then `u.Path, _ = url.JoinPath(u.Path, p)` to finally return `u.String()` but opt for simple fix for this patch.